File: sdkwork-app-sdk-swift/Sources/API/PhoneApi.swift
import Foundation

public class PhoneApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// 手机号验证码登录
    public func login(body: PhoneLoginForm) async throws -> PlusApiResultLoginVO? {
        let response = try await client.post(ApiPaths.appPath("/auth/phone/login"), body: body)
        return response as? PlusApiResultLoginVO
    }
}

File: sdkwork-app-sdk-csharp/Api/PhoneApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using App.Http;
using App.Models;

namespace App.Api
{
    public class PhoneApi
    {
        private readonly HttpClient _client;

        public PhoneApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// 手机号验证码登录
        /// </summary>
        public async Task<PlusApiResultLoginVO?> LoginAsync(PhoneLoginForm body)
        {
            return await _client.PostAsync<PlusApiResultLoginVO>(ApiPaths.AppPath("/auth/phone/login"), body);
        }
    }
}
